fe/fe-core/src/main/java/com/starrocks/planner/PlanNode.java

```diff
@@ -710,16 +712,132 @@ public boolean canPushDownRuntimeFilter() {
     public void checkRuntimeFilterOnNullValue(RuntimeFilterDescription description, Expr probeExpr) {
     }
 
-    public boolean pushDownRuntimeFilters(RuntimeFilterDescription description, Expr probeExpr) {
+    /**
+     * Return candidate slot exprs which is same to the expr, eg: tb1 join tb2 on tb1.a = tb2.b, when expr param is tb1.a,
+     * tb2.b is the candidate slot expr for tb1.a which has the same syntax for the query.
```

Comment:
why tb2.b is the candidate slot expr for tb1.a ?

Reply:
eg: If we push down HJ's t0.a into HJ2, which HJ2's equal conjuncts is  `t1.a=t2.b`, when `t0.a` is `t1.a` so we can also push down `t2.b` because `t1.a` and `t2.b` are equal conjuncts.

But we can only push t1.a  down for t1 and push t2.b for t2. In original codes, we will try all `t1.a` and `t2.b` for both sides.
```
    HJ1
  /     \
HJ2     t0
 /   \
t1  t2
 ```